non-exhaustive patterns/ does not compile

## Сompile time error :
```sh
➜  new-rtsp-service git:(main) ✗ cargo run
   Compiling tracing-gstreamer v0.3.0
   Compiling gstreamer-rtsp-server v0.18.0
error[E0004]: non-exhaustive patterns: `(&Kind(0_u8), _)` and `(&Kind(3_u8..=u8::MAX), _)` not covered
   --> /home/***/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-gstreamer-0.3.0/src/callsite.rs:152:38
    |
152 |             .then_with(move || match (&self.kind, &other.kind) {
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^ patterns `(&Kind(0_u8), _)` and `(&Kind(3_u8..=u8::MAX), _)` not covered
    |
    = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
    = note: the matched value is of type `(&Kind, &Kind)`
```
### Cargo.toml:
```toml
[dependencies]
tracing = "0.1"
tracing-subscriber = "0.3"
tracing-gstreamer = "0.3"
log-panics = "2"


gstreamer-rtsp-server = "0.18"

url = "2"

tokio = {version = "1", features = ["rt-multi-thread", "macros", "sync"]}
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "tracing-gstreamer"
-version = "0.3.1"
+version = "0.3.2"
 edition = "2018"
 authors = ["Simonas Kazlauskas <tracing-gstreamer@kazlauskas.me>"]
 license = "MIT OR Apache-2.0"
@@ -20,13 +20,13 @@ crate-type = ["cdylib", "rlib"]
 libc = "0.2.69"
 once_cell = "1.8.0"
 tracing = "0.1.0"
-tracing-core = ">=0.1.17, <=0.1.21"
+tracing-core = "0.1.17"
 gstreamer = "0.18.0"
 thread_local = "1.0.0"
 
 [dev-dependencies]
 tracing = "0.1.0"
-tracing-subscriber = "0.2.0"
+tracing-subscriber = "0.2.4"
 tracing-tracy = "0.7.0"
 
 [package.metadata.docs.rs]

--- a/src/callsite.rs
+++ b/src/callsite.rs
@@ -9,6 +9,12 @@ use std::{
 use tracing::{field::FieldSet, Level};
 use tracing_core::{identify_callsite, Callsite, Interest, Kind, Metadata};
 
+#[derive(PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Debug)]
+pub(crate) enum GstCallsiteKind {
+    Event = 0,
+    Span = 1,
+}
+
 pub(crate) struct GstCallsite {
     interest: AtomicUsize,
     metadata: Metadata<'static>,
@@ -67,7 +73,10 @@ impl GstCallsite {
                     key.line,
                     key.module.map(|_| &string[..module.len()]),
                     fieldset,
-                    key.kind.clone(),
+                    match key.kind {
+                        GstCallsiteKind::Span => Kind::SPAN,
+                        GstCallsiteKind::Event => Kind::EVENT,
+                    },
                 ),
             });
             &*callsite
@@ -123,7 +132,7 @@ struct Key<'a> {
     target: &'a str,
     name: &'static str,
     fields: &'static [&'static str],
-    kind: Kind,
+    kind: GstCallsiteKind,
 }
 
 impl PartialEq for Key<'_> {
@@ -149,13 +158,7 @@ impl Ord for Key<'_> {
             .then_with(move || self.file.cmp(&other.file))
             .then_with(move || self.name.cmp(other.name))
             .then_with(move || self.fields.cmp(other.fields))
-            .then_with(move || match (&self.kind, &other.kind) {
-                (&Kind::EVENT, &Kind::EVENT) | (&Kind::SPAN, &Kind::SPAN) => {
-                    std::cmp::Ordering::Equal
-                }
-                (&Kind::EVENT, &Kind::SPAN) => std::cmp::Ordering::Less,
-                (&Kind::SPAN, &Kind::EVENT) => std::cmp::Ordering::Greater,
-            })
+            .then_with(move || self.kind.cmp(&other.kind))
             .then_with(move || self.target.cmp(other.target))
     }
 }
@@ -177,7 +180,7 @@ impl DynamicCallsites {
         file: Option<&str>,
         module: Option<&str>,
         line: Option<u32>,
-        kind: Kind,
+        kind: GstCallsiteKind,
         fields: &'static [&'static str],
     ) -> &'static GstCallsite {
         let mut guard = self.data.lock().unwrap_or_else(PoisonError::into_inner);
@@ -188,7 +191,7 @@ impl DynamicCallsites {
             file,
             module,
             line,
-            kind: kind.clone(),
+            kind,
             fields,
         };
         if let Some(callsite) = guard.get(&lookup_key) {

--- a/src/log.rs
+++ b/src/log.rs
@@ -1,3 +1,4 @@
+use crate::{callsite::GstCallsiteKind, state_desc, PadFlags};
 use gstreamer::{
     ffi::{
         gst_debug_add_log_function, gst_debug_category_get_name, gst_debug_message_get,
@@ -12,9 +13,7 @@ use gstreamer::{
 };
 use libc::{c_char, c_int, c_void};
 use std::{convert::TryFrom, ffi::CStr};
-use tracing_core::{Callsite, Event, Kind, Level};
-
-use crate::{state_desc, PadFlags};
+use tracing_core::{Callsite, Event, Level};
 
 unsafe extern "C" fn log_callback(
     category: *mut gstreamer::ffi::GstDebugCategory,
@@ -71,7 +70,7 @@ unsafe extern "C" fn log_callback(
             Some(&file),
             Some(&module),
             Some(line as u32),
-            Kind::EVENT,
+            GstCallsiteKind::Event,
             &[
                 "message",
                 "gobject.address",

--- a/src/tracer.rs
+++ b/src/tracer.rs
@@ -1,5 +1,4 @@
-use std::cell::RefCell;
-
+use crate::callsite::GstCallsiteKind;
 use glib::subclass::basic;
 use gstreamer::{
     glib,
@@ -8,8 +7,8 @@ use gstreamer::{
     traits::{GstObjectExt, PadExt},
     Buffer, FlowReturn, Object, Pad, Query, Tracer,
 };
+use std::cell::RefCell;
 use tracing::{span::Attributes, Callsite, Dispatch, Id};
-use tracing_core::Kind;
 
 struct EnteredSpan {
     id: Id,
@@ -51,7 +50,7 @@ impl TracingTracerPriv {
             None,
             None,
             None,
-            Kind::SPAN,
+            GstCallsiteKind::Span,
             &["gstpad.state", "gstpad.parent.name"],
         );
         let interest = callsite.interest();

